### ISSUE TITLE
BTDB Tracker Failing in Sonarr and Radarr - Jackett Version 0.8.225.0

### ISSUE THREAD

**Issue by OasisFounder** [Jackett/Jackett#1900]
Logs are as follows from Sonarr:

HTTP Error - Res: [GET] http://127.0.0.1:9117/api/v2.0/indexers/btdb/results/torznab/api?t=tvsearch&cat=5030,5040&extended=1&apikey=(removed)&offset=0&limit=100: 400.BadRequest
<Error><Message>btdb does not support the requested query.</Message></Error>

Unable to connect to indexer: HTTP request failed: [400:BadRequest] [GET] at [http://127.0.0.1:9117/api/v2.0/indexers/btdb/results/torznab/api?t=tvsearch&cat=5030,5040&extended=1&apikey=8wfwftxlba4hbbv9abnxb0hktvy0hdwp&offset=0&limit=100]

Exception

NzbDrone.Common.Http.HttpException: HTTP request failed: [400:BadRequest] [GET] at [http://127.0.0.1:9117/api/v2.0/indexers/btdb/results/torznab/api?t=tvsearch&cat=5030,5040&extended=1&apikey=8wfwftxlba4hbbv9abnxb0hktvy0hdwp&offset=0&limit=100]
   at NzbDrone.Common.Http.HttpClient.Execute(HttpRequest request) in M:\BuildAgent\work\b69c1fe19bfc2c38\src\NzbDrone.Common\Http\HttpClient.cs:line 94
   at NzbDrone.Core.Indexers.HttpIndexerBase`1.FetchIndexerResponse(IndexerRequest request) in M:\BuildAgent\work\b69c1fe19bfc2c38\src\NzbDrone.Core\Indexers\HttpIndexerBase.cs:line 302
   at NzbDrone.Core.Indexers.HttpIndexerBase`1.FetchPage(IndexerRequest request, IParseIndexerResponse parser) in M:\BuildAgent\work\b69c1fe19bfc2c38\src\NzbDrone.Core\Indexers\HttpIndexerBase.cs:line 288
   at NzbDrone.Core.Indexers.HttpIndexerBase`1.TestConnection() in M:\BuildAgent\work\b69c1fe19bfc2c38\src\NzbDrone.Core\Indexers\HttpIndexerBase.cs:line 315
<Error><Message>btdb does not support the requested query.</Message></Error>

Invalid request Validation failed: 
 -- Unable to connect to indexer, check the log for more details

---------

Radarr also fails providing these errors in the log:

Unable to connect to indexer: Sequence contains no elements

Exception

System.InvalidOperationException: Sequence contains no elements
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
   at NzbDrone.Core.Indexers.HttpIndexerBase`1.TestConnection() in C:\projects\radarr-usby1\src\NzbDrone.Core\Indexers\HttpIndexerB

Invalid request Validation failed: 
 -- Unable to connect to indexer, check the log for more details

**Comment by garfield69** [comment]
try changing your categories in your Sonarr BTDB indexer definition to 2000,5000 and try again.

**Comment by OasisFounder** [comment]
Ok - sonar seems to work ok using those categories but still having issues with Radarr?

**Comment by garfield69** [comment]
I've researched this and found the problem.     

The BTDB site does not use any categories, so in order for the btdb definition to work properly with Sonarr and Radarr, we have to provide a hard-coded category to be included in the returned results.    
 
I've decided that category `Other` is the best fit, since BTDB contains magnets for all sorts of content.      
So once #1921 is accepted, the next release of Jackett will include a revised BTDB definition.     

To use BTDB on either Sonarr or Radarr, create/amend your torznab indexer setup category to `100001`      
I've tested this definition on the Jackett Dashboard, Sonarr for a TV search, and Radarr for a movie search.
